test/helpers/cmd.go

```diff
@@ -317,6 +352,71 @@ func (res *CmdRes) WaitUntilMatchRegexp(expr string, timeout time.Duration) erro
 		&TimeoutConfig{Timeout: timeout})
 }
 
+// WaitUntilMatchFilterLineTimeout applies the JSONPath 'filter' to each line of
+// `CmdRes.stdout` and waits until a line matches the 'expected' output.
+// If the 'timeout' is reached it will return an error.
+func (res *CmdRes) WaitUntilMatchFilterLineTimeout(filter, expected string, timeout time.Duration) error {
+	parser := jsonpath.New("").AllowMissingKeys(true)
+	err := parser.Parse(filter)
+	if err != nil {
+		return err
+	}
+
+	errChan := make(chan error, 1)
+	body := func() bool {
+		lines := res.ByLines()
+		for i, line := range lines {
```

Comment:
Should this loop be replaced by `FilterLines` call?

Reply:
Removed the code duplication by introducing a shared `filterLinesJSONPath`.